bemani/backend/iidx/cannonballers.py

```diff
@@ -1,15 +1,2184 @@
+                score_list = Node.void('score_list')
+                cpu_score_list.add_child(score_list)
+                percentage = grade_to_percentage_map.get(cpu_list[index], 0)
+                notes = music_list[i]
+                score = (notes * 2 * percentage) // 100
+                ghost_section = score // 64
+                ghost = [ghost_section] * 64
+                ghost_total = ghost_section * 64
+                ghost_lost = score - ghost_total
+                ghost[0] = ghost[0] + ghost_lost
+                score_list.add_child(Node.s32('score', score))
+                score_list.add_child(Node.bool('enable_score', True))
+                score_list.add_child(Node.u8_array('ghost', ghost))
+                score_list.add_child(Node.bool('enable_ghost', True))
+        return root
+
+    def handle_IIDX25grade_raised_request(self, request: Node) -> Node:
+        extid = int(request.attribute('iidxid'))
+        cltype = int(request.attribute('gtype'))
+        rank = self.game_to_db_rank(int(request.attribute('gid')), cltype)
+
+        userid = self.data.remote.user.from_extid(self.game, self.version, extid)
+        if userid is not None:
+            percent = int(request.attribute('achi'))
+            stages_cleared = int(request.attribute('cstage'))
+            cleared = stages_cleared == self.DAN_STAGES
+
+            if cltype == self.GAME_CLTYPE_SINGLE:
+                index = self.DAN_RANKING_SINGLE
+            else:
+                index = self.DAN_RANKING_DOUBLE
+
+            self.update_rank(
+                userid,
+                index,
+                rank,
+                percent,
+                cleared,
+                stages_cleared,
+            )
+
+        # Figure out number of players that played this ranking
+        all_achievements = self.data.local.user.get_all_achievements(self.game, self.version)
+        num_players = 0
+        for [_, ach] in all_achievements:
+            if ach.type != index:
+                continue
+            if ach.id != rank:
+                continue
+            num_players = num_players + 1
+
+        root = Node.void('IIDX25grade')
+        root.set_attribute('pnum', str(num_players))
+        return root
+
+    def handle_IIDX25pc_common_request(self, request: Node) -> Node:
+        root = Node.void('IIDX25pc')
+        root.set_attribute('expire', '600')
+
+        ir = Node.void('ir')
+        root.add_child(ir)
+        ir.set_attribute('beat', '2')
+
+        vip_black_pass = Node.void('vip_pass_black')
+        root.add_child(vip_black_pass)
+
+        newsong_another = Node.void('newsong_another')
+        root.add_child(newsong_another)
+        newsong_another.set_attribute('open', '1')
+
+        deller_bonus = Node.void('deller_bonus')
+        root.add_child(deller_bonus)
+        deller_bonus.set_attribute('open', '1')
+
+        common_evnet = Node.void('common_evnet')  # Yes, this is misspelled in the game
```

Comment:
Still?

Reply:
If there's anything i've learned working on konami games, they don't usually change old code or even get rid of it. The expert mode stuff is still in the net code for newer games and there's even a button in rootage even though the mode is completely removed